tests/src/test/scala/com/nvidia/spark/rapids/ShufflePartitionerRetrySuite.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids
+
+import ai.rapids.cudf.Table
+import com.nvidia.spark.rapids.Arm.withResource
+import com.nvidia.spark.rapids.RapidsPluginImplicits._
+import com.nvidia.spark.rapids.jni.RmmSpark
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.catalyst.expressions.{Ascending, AttributeReference, ExprId, SortOrder, SpecificInternalRow}
+import org.apache.spark.sql.types.{DataType, IntegerType, StringType}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+class ShufflePartitionerRetrySuite extends RmmSparkRetrySuiteBase {
+  private def buildBatch(): ColumnarBatch = {
+    withResource(new Table.TestBuilder()
+      .column(9, null.asInstanceOf[java.lang.Integer], 8, 7, 6, 5, 4, 3, 2, 1)
+      .column("nine", "eight", null, null, "six", "five", "four", "three", "two", "one")
+      .build()) { table =>
+      GpuColumnVector.from(table, Array(IntegerType, StringType))
+    }
+  }
+
+  private def testRoundRobinPartitioner(partNum: Int) = {
+    TestUtils.withGpuSparkSession(new SparkConf()) { _ =>
+      val rrp = GpuRoundRobinPartitioning(partNum)
+      // batch will be closed within columnarEvalAny
+      val batch = buildBatch
```

Comment:
nit: it might be nice to build the batch right when we call `columnarEvalAny` just so there is less code between where it is created and where it is consumed. Just because it could leak if there is an exception inbetween.

Reply:
It's separated due to ```buildBatch``` method implement didn't have a retry in this test suite. We can do it and since it's just a test, so I skipped it.